rllib/env/testing/single_agent_gym_env_runner.py

```diff
@@ -0,0 +1,143 @@
+from typing import List, Optional, Tuple
+
+import gymnasium as gym
+
+from ray.rllib.algorithms.algorithm_config import AlgorithmConfig
+from ray.rllib.env.env_runner import EnvRunner
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.replay_buffers.episode_replay_buffer import _Episode as Episode
+
+
+class SingleAgentGymEnvRunner(EnvRunner):
+    """A simple single-agent EnvRunner subclass for testing purposes.
+
+    Uses a gym.vector.Env environment and random actions.
+    """
+
+    def __init__(self, *, config: AlgorithmConfig, **kwargs):
+        """Initializes a SimpleEnvRunner instance.
+
+        Args:
+            config: The config to use to setup this EnvRunner.
+        """
+        super().__init__(config=config, **kwargs)
+
+        # Create the gym.vector.Env object.
+        self.env = gym.vector.make(
+            self.config.env,
+            num_envs=self.config.num_envs_per_worker,
+            asynchronous=self.config.remote_worker_envs,
+            **dict(self.config.env_config, **{"render_mode": "rgb_array"}),
+        )
+        self.num_envs = self.env.num_envs
+
+        self._needs_initial_reset = True
+        self._episodes = [None for _ in range(self.num_envs)]
+
+    @override(EnvRunner)
+    def sample(
+        self,
+        *,
+        num_timesteps: Optional[int] = None,
+        num_episodes: Optional[int] = None,
+        explore: bool = True,
+        random_actions: bool = False,
+        force_reset: bool = False,
+        **kwargs,
+    ) -> Tuple[List[Episode], List[Episode]]:
```

Comment:
Can you add a docstring on what the expected output type is? and what each element in the tuple represent?